modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2164,7 +2167,7 @@ void findBroadAxis(const MatShape& broadShape, const MatShape& outShape, size_t&
 
     // find the first non-one element of the broadcasting shape
     axis = 0;
-    for (; axis < broadShape.size() && broadShape[axis] == 1; ++axis) {}
+    for (; axis < broadShape.size() && broadShape[axis] == 1 && broadShape[axis] == outShape[axis]; ++axis) {}
 
     // find the last non-one element of the broadcasting shape
     size_t endAxis = broadShape.size();
```

Comment:
For new PR in 3.4 branch: I think there might be a symmetric bug in the loop below.

Reply:
Ok, I will check the detailed running logic.